### ISSUE TITLE
Dict typechecking behaviour with no args does not correspond to Dict[Any, Any]

### ISSUE THREAD

**Issue by jtbeach** [agronholm/typeguard#92]
I'm not sure if this behaviour is intended or not, but it does differ from mypy.

If I declare an argument or return value as just `Dict`, I usually intend that to mean `Dict[Any, Any]`. This seems to correspond to what mypy is doing.

What typeguard seems to do is determine the type of the keys and values from the first item in the dict if the types are initially unbound.

The practical implication of this is that unless all your values are of the same type, you need to explicitly specify to type as `Dict[str, Any]`. This is a very common issue in parsed JSON for example where some values are int, lists dicts, etc.

Just wondering if this is intended behaviour or a bug? It certainly makes writing type annotations unnecessarily verbose IMO.

**Comment by agronholm** [comment]
How can I reproduce this? I tried the following and it gave me no errors:
```python3
from typing import Dict

from typeguard import typechecked


@typechecked
def check(x: Dict):
    pass

check({'a': 3, 7: True})
```

**Comment by jtbeach** [comment]
```

**Comment by jtbeach** [comment]
Interestingly, I also get failures on your test program

**Comment by agronholm** [comment]
I was able to reproduce the problem once I switched to Python 3.7. I think I understand now – the type variables from the original definition are applied here because the only way to distinguish between `Dict` using custom type variables and the original is to do an identity check against the module of origin. This did not used to be the case before Python 3.7. I've had to deal with this same problem in my other project ([sphinx-autodoc-typehints](https://github.com/agronholm/sphinx-autodoc-typehints)).
